### DISCUSSION TITLE
Acceptable Poll Limits

### DISCUSSION THREAD

**Discussion by DanubeRS** [amberelectric/public-api#55] [Q&A] [answered]
Hey @madpilot 

What sort of rates are acceptable for hitting the endpoints?

I used to cache values based on the returned 5-minute window of the old api, but this seems a little trickier (given the 5-min windows are not reported for `CurrentInterval`).

My short-term solution is to poll with increased frequency, but I assume this will go away with consolidated 5MS in the coming months.

**Comment by madpilot** [comment, marked as answer]
Every minute is fine. We'll be formally enforcing a limit shortly, but it will be more than enough to get the data required.

Just be aware: the old API didn't return the raw 5-minute interval. See the discussion here: https://github.com/amberelectric/public-api/discussions/38

**Reply by DanubeRS** [reply]
Great, I have set my refresh to around this time.

And yeah, thats more or less what I have concluded the old API was, an average of the cleared 5m intervals. Convergence patterns towards 30m were somewhat predictable, but I'm glad the forecasts (despite being pretty loose, due to market behaviour) are included now.

As an aside, how often are forecasts updated? As the bid stack changes? On each block progressing? More for curiosity, as the granularity is mostly useless for the average user.

**Reply by madpilot** [reply]
We poll AEMO once a minute, so it's as often as they change them. The `ForecastIntervals` are 30-minute forecasts, so they seem to update more-or-less every 30 minutes. I don't know the answer to your actual question of how often AEMO updates them, though.

At around midday each day, the next 24-hours of forecasts get released, so the number of Forecasts we can return is between 24 and 72 intervals.
